xpsi/surface_radiation_field/__init__.pyx

```diff
@@ -180,33 +208,47 @@ def intensity(double[::1] energies,
         units of photons/s/keV/cm^2/sr.
 
     """
+    cdef size_t _beam_opt = beam_opt
+    cdef size_t _atmos_extension
+
     cdef fptr_init init_ptr = NULL
     cdef fptr_free free_ptr = NULL
     cdef fptr_eval eval_ptr = NULL
     cdef fptr_norm norm_ptr = NULL
 
-    if extension == 'hot':
+    if region_extension == 'hot':
         init_ptr = init_hot
         free_ptr = free_hot
         eval_ptr = eval_hot
         norm_ptr = eval_hot_norm
-    elif extension == 'elsewhere':
+    elif region_extension == 'elsewhere':
         init_ptr = init_elsewhere
         free_ptr = free_elsewhere
         eval_ptr = eval_elsewhere
         norm_ptr = eval_elsewhere_norm
     else:
-        raise ValueError("Extension module must be 'hot' or 'elsewhere'.")
+        raise ValueError("Region extension module must be 'hot' or 'elsewhere'.")
```

Comment:
Naive question:  shouldn't the `region_extension` also allow `everywhere` as well ?

Reply:
I think not. There never used to be a separate atmosphere extension for everywhere (it has just used the one of hot region). The purpose of this function is to test atmosphere extensions, so testing it using "hot" already tests it for "everywhere". Maybe, if want to be super clear, we could allow region_extension=everywhere here but it would then just need to do the same as region_extension=hot.